### ISSUE TITLE
Q. Training and Evaluating on Dataset.

### ISSUE THREAD

**Issue by opencomvis** [lightly-ai/lightly#1284]
Hi! Thanks for your amazing work. 
_I have a question that's related to SSL, not your code._ 
Is it possible to train a model using SSL on all samples of a dataset then use the pretrainded model on a downstream task on the same dataset? or should I split the dataset to samples for pretraining and others for the downstream task? 
_Another quesion_, when I train SSL on my model, projections heads will be added, so, should we remove these layers when using the model for the downstream task? 
Thank you.

**Comment by philippmwirth** [comment]
Hi @opencomvis, great question! We should add a best practice section to the docs...

> Is it possible to train a model using SSL on all samples of a dataset then use the pretrainded model on a downstream task on the same dataset? or should I split the dataset to samples for pretraining and others for the downstream task?

This is definitely possible and done often in practice. For example if you have 1000 images with 100 annotations, you can do self-supervised pre-training on the 1000 and then use the 100 for finetuning. Only caveat: Don't include the test set in your pre-training data.

> Another quesion, when I train SSL on my model, projections heads will be added, so, should we remove these layers when using the model for the downstream task?

Typically you want to use only the backbone for downstream tasks (il.e. discard prediction and projection heads).

**Comment by opencomvis** [comment]
Thank you for your reply. One more question please: is it possible to add a monitor for the features learnt ? and if yes, do you have an example ?, such a classifier excuted each epoch on a downstream task to evaluate the quality of features extracted so far.

**Comment by philippmwirth** [comment]
> Thank you for your reply. One more question please: is it possible to add a monitor for the features learnt ? and if yes, do you have an example ?, such a classifier excuted each epoch on a downstream task to evaluate the quality of features extracted so far.

You can take a look at the models implemented in `benchmarks/imagenet/resnet50/` for an example of an online classifier.

**Comment by opencomvis** [comment]
Thank you
